components/automate-ui/src/app/components/notification/notification.component.spec.ts

```diff
@@ -0,0 +1,142 @@
+import { MockComponent } from 'ng2-mock-component';
+import { using } from 'app/testing/spec-helpers';
+
+import { ChefNotificationComponent } from './notification.component';
+import { Type } from 'app/entities/notifications/notification.model';
+
+describe('ChefNotificationComponent', () => {
+  let component: ChefNotificationComponent;
+  let fixture: ComponentFixture<ChefNotificationComponent>;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      imports: [ BrowserAnimationsModule ],
+      declarations: [
+        ChefNotificationComponent,
+        MockComponent({ selector: 'chef-icon' })
+      ],
+      schemas: [ CUSTOM_ELEMENTS_SCHEMA ]
+    })
+      .compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(ChefNotificationComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  const genNotification = (type: Type, timeout: number) => {
+    component.type = type;
+    component.timeout = timeout;
+  };
+
+  describe('generates a notification', () => {
+
+    it('and defaults to type of info and timeout of 8', () => {
+      expect(component.type).toBe('info');
+      expect(component.timeout).toEqual(8);
+    });
+
+    using([
+      [Type.info, 10],
+      [Type.info, 7],
+      [Type.info, 4],
+      [Type.info, 2],
+      [Type.info, 0],
+      [Type.error, 43],
+      [Type.error, 13],
+      [Type.error, 3],
+      [Type.error, 23]
+    ], function (type: Type, timeout: number) {
+          it('and reads type and timeout from inputs', () => {
+            genNotification(type, timeout);
+
+            expect(component.type).toBe(type);
+            expect(component.timeout).toEqual(timeout);
+          });
+        });
+  });
+
+  describe('ngAfterViewInit', () => {
+
+    using([
+      [Type.info, 10, false],
+      [Type.info, 0, true],
+      [Type.info, 4, false],
+      [Type.info, 2, false],
+      [Type.info, 0, true],
+      [Type.error, 43, false],
+      [Type.error, 0, true],
+      [Type.error, 7, false]
+    ], function (type: Type, timeout: number, isInfinite: boolean) {
+      it('determinites notification timeout is inifinite when timeout equals 0', () => {
+        genNotification(type, timeout);
+        component.ngAfterViewInit();
+
+        expect(component.isInfinite).toBe(isInfinite);
+      });
+    });
+
+    using([
+      [Type.info, 1, false],
+      [Type.info, 2, false],
+      [Type.error, 3, false]
+    ], function (type: Type, timeout: number, isInfinite: boolean) {
+      it('notification should close when infinite', fakeAsync(() => {
+        spyOn(component, 'handleClose');
+        genNotification(type, timeout);
+        component.ngAfterViewInit();
+
+        expect(component.isInfinite).toBe(isInfinite);
+        tick((timeout * 1000) + 1);
```

Comment:
I am not well-versed on the `fakeAsync`, but I believe these `tick` calls are using real-time, which is a major red flag. That slows down the unit tests by a huge factor; any given test file should run in milliseconds, not seconds. (Sometimes time-related testing is where you do need to test implementation. 😁 ) Found one article that (rather teresely) indicates that `flush` is sufficient and `tick` may not be needed: https://alligator.io/angular/testing-async-fakeasync/

Reply:
This was a great quick read -thank you!